Bug: `jsonStringify` produces trailing comma if replacer omits the last object property

### Bug description

When using `jsonStringify` with a custom replacer function, if the replacer omits the last property of an object (by returning `undefined`), the generated string contains a trailing comma (making it invalid JSON).

```

import { jsonStringify } from "./lib/common.js";

const obj = { a: "foo", b: "bar" };
const replacer = (key, value) => key === "b" ? undefined : value;

console.log(jsonStringify(obj, replacer));
// Output: '{"a":"foo",}' (Invalid JSON)

```

### Expected behavior
It should match standard JSON.stringify(obj, replacer) output:

```
'{"a":"foo"}'
```

### Failing Test Case
Adding the following test case inside the `replacer` block of `lib/stringify.spec.ts` reproduces the failure:
```
it("should remove the last property when it returns undefined without leaving a trailing comma", () => {
  const value = {
    aaa: "foo",
    bbb: "bar"
  };
  const replacer = (key: string, value: unknown) => {
    if (key !== "bbb") {
      return value;
    }
  };

  expect(jsonStringify(value, replacer)).to.eql(JSON.stringify(value, replacer));
});

```

**Vitest Error Output:**
```
AssertionError: expected '{"aaa":"foo",}' to deeply equal '{"aaa":"foo"}'
```

--- a/lib/common.js
+++ b/lib/common.js
@@ -126,15 +126,16 @@ const stringifyObject = (value, replacer, space, pointer, depth) => {
   const colonSpacing = space ? " " : "";
 
   let result = "{" + padding + space;
-  for (let index = 0; index < entries.length; index++) {
-    const [key, value] = entries[index];
+  let first = true;
+  for (const [key, value] of entries) {
     const keyPointer = JsonPointer.append(key, pointer);
     const stringifiedValue = stringifyValue(value, replacer, space, key, keyPointer, depth + 1);
     if (stringifiedValue !== undefined) {
-      result += JSON.stringify(key) + ":" + colonSpacing + stringifiedValue;
-      if (entries[index + 1]) {
+      if (!first) {
         result += `,${padding}${space}`;
       }
+      first = false;
+      result += JSON.stringify(key) + ":" + colonSpacing + stringifiedValue;
     }
   }
   return result + padding + "}";

--- a/lib/stringify.spec.ts
+++ b/lib/stringify.spec.ts
@@ -39,6 +39,20 @@ describe("Json.stringify", () => {
       expect(jsonStringify(value, replacer)).to.eql(JSON.stringify(value, replacer));
     });
 
+    it("should remove the last property when it returns undefined without leaving a trailing comma", () => {
+      const value = {
+        aaa: "foo",
+        bbb: "bar"
+      };
+      const replacer = (key: string, value: unknown) => {
+        if (key !== "bbb") {
+          return value;
+        }
+      };
+
+      expect(jsonStringify(value, replacer)).to.eql(JSON.stringify(value, replacer));
+    });
+
     it("should remove items that return undefined", () => {
       const value = ["foo", "bar"];
       const replacer = (key: string, value: unknown) => {

